template/asyncapi.js

```diff
@@ -0,0 +1,555 @@
+  const anyOf = prop.anyOf() && prop.anyOf().map((p, idx) => (
+    <SchemaProp prop={p} propName={idx} path={propName || buildPath(path, idx)} />
+  ));
+  const allOf = prop.allOf() && prop.allOf().map((p, idx) => (
+    <SchemaProp prop={p} propName={idx} path={propName || buildPath(path, idx)} />
+  ));
+  const oneOf = prop.oneOf() && prop.oneOf().map((p, idx) => (
+    <SchemaProp prop={p} propName={idx} path={propName || buildPath(path, idx)} />
+  ));
+  const properties = Object.entries(prop.properties()).map(([pName, p]) => {
+    const circProps = p.circularProps();
+    const isPropCircular = circularPropsParent.includes(pName);
+    
+    if (isPropCircular === true) {
+      return (
+        <SchemaPropRow
+          prop={p} 
+          propName={pName}
+          path={propName || buildPath(path, pName)} 
+          required={isRequired(prop, pName)}
+          isCircular={isPropCircular}
+        />
+      );
+    } else {
+      return (
+        <SchemaProp 
+          prop={p} 
+          propName={pName} 
+          path={propName || buildPath(path, pName)} 
+          required={isRequired(prop, pName)}
+          circularPropsParent={circProps}
+        />
+      );
+    }
+  });
+  const additionalProperties = prop.additionalProperties() && prop.additionalProperties().properties 
+    ? Object.entries(prop.additionalProperties()).map(([pName, p]) => (
+      <SchemaProp prop={p} propName={pName} path={propName || buildPath(path, pName)} required={isRequired(prop.additionalProperties(), pName)} />
+    )) : null;
+  const items = prop.items() && prop.items().properties
+    ? Object.entries(prop.items().properties()).map(([pName, p]) => {
+      const isCirc = p.isCircular();
+
+      if (isCirc === true) {
+        return (
+          <SchemaPropRow
+            prop={p} 
+            propName={pName}
+            path={propName || buildPath(path, pName)} 
+            required={isRequired(prop, pName)}
+            isCircular={isCirc}
+          />
+        );
+      } else {
+        return (
+          <SchemaProp 
+            prop={p} 
+            propName={pName} 
+            path={propName || buildPath(path, pName)} 
+            required={isRequired(prop.items(), pName)}
+          />
+        );
+      }
+    }) : null;
+
+  return (
+    <>
+      <SchemaPropRow prop={prop} propName={propName} required={required} path={path} />
+      {anyOf}
+      {allOf}
+      {oneOf}
+      {properties}
+      {additionalProperties}
+      {items}
+    </>
+  );
+}
+
+function SchemaPropRow({ prop, propName, required = false, path = '', isCircular = false }) {
+  const acceptedValues = prop.enum() && prop.enum().length ? prop.enum().join(', ') : '_Any_';
+  const types = `${prop.anyOf() ? `anyOf` : ''}${prop.allOf() ? `allOf` : ''}${prop.oneOf() ? `oneOf` : ''}${prop.items() && prop.items().type() ? `${prop.items().type()}` : ''}`;
```

Comment:
This fails when trying to run the dummy example from the [generator](https://github.com/asyncapi/generator/blob/master/test/docs/dummy.yml). 

Did you mean something like:
```suggestion
  const types = `${prop.anyOf() ? `anyOf` : ''}${prop.allOf() ? `allOf` : ''}${prop.oneOf() ? `oneOf` : ''}${prop.items() && !Array.isArray(prop.items()) && prop.items().type() ? `${prop.items().type()}` : ''}`;
```

Reply:
Can we make this code a bit more legible? Maybe a switch/case would be better in this case.

Reply:
@jonaslagoni Thanks for report! I see that stack trace also works perfectly!
![image](https://user-images.githubusercontent.com/20404945/101283579-88f42800-37db-11eb-827a-49a6efccbbcb.png)

@fmvilas I will try to simplify it :)